server.py

```diff
@@ -43,31 +55,32 @@
         statement = fl.read()
     questions[i] = Question(output=output, statement=statement)
 
+
 @app.route("/")
 def changePath():
     return redirect("/dashboard")
 
+
 @app.get("/dashboard")
 def dashboard():
     return template("dashboard.html", contests=contests)
 
+
 @app.get("/contest/<code>/<number>")
-def contest(code,number):
+def contest(code, number):
     statement = questions[number].statement
-    return template("index.html", question_number=number, question=statement)
+    return template(
+        "index.html", question_number=number, contest=code, question=statement
+    )
+
 
 @app.get("/contest/<code>")
 def contest(code):
-    for contest in contests:
-        if(contest.code == code):
-            description = contest.description
-            questions = contest.questions
-    return template("contest.html", code=code, description=description, questions=questions)
-
-@app.get("/question/<number>")
-def question(number):
-    statement = questions[number].statement
-    return template("index.html", question_number=number, question=statement)
+    if not code in contests:
+        return "Contest does not exist"
+    if contests[code].start_time > datetime.datetime.now():
```

Comment:
we shouldn't be able to attempt after a contest is closed

Reply:
Most online Judges keep contest questions open to attempt after the contest.
The score for submissions after the contest ends is not considered in the ranking.
I kept it the same way. We could display a message that contest has ended. Or do you want me to close the contest after its time is over.

Reply:
sure sure. keep it open but don't record the score then? The code submission api needs a change I think

Reply:
We are recording just the submissions to problems. The score is calculated while displaying the ranking and it calculates the score only for the timings the contest is on

Reply:
This one's fine then, I guess